File: blog-sever/server-jpa/src/main/kotlin/com/site/blog/model/entity/Link.kt
package com.site.blog.model.entity

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import java.io.Serializable
import java.sql.Timestamp

/**
 *
 *
 * 友情链接表
 *
 *
 */
@Entity
data class Link  (
    /**
     * 友链表主键id
     */


    /**
     * 友链类别 0-友链 1-推荐 2-个人网站
     */
    
    var linkType: Int? = null,

    /**
     * 网站名称
     */
    
    var linkName: String? = null,

    /**
     * 网站链接
     */
    
    var linkUrl: String? = null,

    /**
     * 网站描述
     */
    
    var linkDescription: String? = null,

    /**
     * 用于列表排序
     */
    
    var linkRank: Int? = null,

    /**
     * 是否删除 0-未删除 1-已删除
     */
    
    var show: Boolean? = null,

    /**
     * 添加时间
     */
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    
    var createTime: Timestamp? = null,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
    
    var updateTime: Timestamp? = null,


) {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "link_id", nullable = false)
    open var linkId: Long? = null
}

File: BlogApi/BlogApi/Models/Link.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BlogApi.Models;

[Table("link")]
public class Link
{
    [Key]
    [Column("link_id")]
    public string LinkId { get; set; } = Guid.NewGuid().ToString("N");

    [Column("link_type")]
    public short LinkType { get; set; }

    [Column("link_name")]
    public string LinkName { get; set; } = string.Empty;

    [Column("link_url")]
    public string LinkUrl { get; set; } = string.Empty;

    [Column("link_description")]
    public string LinkDescription { get; set; } = string.Empty;

    [Column("link_rank")]
    public int LinkRank { get; set; }

    [Column("show")]
    public bool Show { get; set; }

    [Column("create_time")]
    public DateTime CreateTime { get; set; }

    [Column("update_time")]
    public DateTime? UpdateTime { get; set; }
}
